domador.js

```diff
@@ -490,7 +490,11 @@ Domador.prototype.process = function process (el) {
     case 'PRE':
       if (this.options.fencing) {
         this.append('\n\n');
-        this.output('```' + (this.options.fencinglanguage(el) || '') + '\n');
+        var fencinglanguage = this.options.fencinglanguage(el);
```

Comment:
Let's extract this whole logic into a separate function, as it's getting out of hand.